cmd/golangBuild.go

```diff
@@ -0,0 +1,262 @@
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperenv"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+const coverageFile = "cover.out"
+
+type golangBuildUtils interface {
+	command.ExecRunner
+
+	FileExists(filename string) (bool, error)
+	FileRead(path string) ([]byte, error)
+	FileWrite(path string, content []byte, perm os.FileMode) error
+
+	// Add more methods here, or embed additional interfaces, or remove/replace as required.
+	// The golangBuildUtils interface should be descriptive of your runtime dependencies,
+	// i.e. include everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type golangBuildUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to golangBuildUtils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// golangBuildUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newGolangBuildUtils() golangBuildUtils {
+	utils := golangBuildUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func golangBuild(config golangBuildOptions, telemetryData *telemetry.CustomData) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := newGolangBuildUtils()
+
+	// Error situations will be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runGolangBuild(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("execution of golang build failed")
+	}
+}
+
+func runGolangBuild(config *golangBuildOptions, telemetryData *telemetry.CustomData, utils golangBuildUtils) error {
+
+	// install test pre-requisites only in case testing should be performed
+	if config.RunTests || config.RunIntegrationTests {
+		if err := utils.RunExecutable("go", "install", "gotest.tools/gotestsum@latest"); err != nil {
+			return fmt.Errorf("failed to install pre-requisite: %w", err)
+		}
+	}
+
+	failedTests := false
+
+	if config.RunTests {
+		success, err := runGolangTests(config, utils)
+		if err != nil {
+			return err
+		}
+		failedTests = !success
+	}
+
+	if config.RunTests && config.ReportCoverage {
+		if err := reportGolangTestCoverage(config, utils); err != nil {
+			return err
+		}
+	}
+
+	if config.RunIntegrationTests {
+		success, err := runGolangIntegrationTests(config, utils)
+		if err != nil {
+			return err
+		}
+		failedTests = !success
```

Comment:
We overwrite the result from test execution here.
```suggestion
		failedTests = failedTests || !success
```

Reply:
good catch!

Reply:
done